lnrpc/routerrpc/router_backend.go

```diff
@@ -460,7 +467,26 @@ func UnmarshallHopWithPubkey(rpcHop *lnrpc.Hop, pubkey route.Vertex) (*route.Hop
 		LegacyPayload:    false,
 		MPP:              mpp,
 		AMP:              amp,
-	}, nil
+		EncryptedData:    rpcHop.EncryptedData,
+		TotalAmtMsat:     lnwire.MilliSatoshi(rpcHop.TotalAmtMsat),
+	}
+
+	haveBlindingPoint := len(rpcHop.BlindingPoint) != 0
+	if haveBlindingPoint {
+		hop.BlindingPoint, err = btcec.ParsePubKey(
+			rpcHop.BlindingPoint,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("blinding point: %w", err)
+		}
+	}
+
+	if haveBlindingPoint && len(rpcHop.EncryptedData) == 0 {
```

Comment:
Q: if we only have one blinded hop, do we still need this `EncryptedData` field?

Reply:
We still need it for a single hop blinded path because it allows the recipient to verify that they originally created the blinded path (and are not being probed). 

The payload will contain a `path_id` that the introduction node/recipient will check against its invoices (kind of like a `payment_secret` in today's world).